tests/rbd_mirror/test_rbd_mirror_image_migration.py

```diff
@@ -0,0 +1,262 @@
+            dest_ec_spec = dest_ec_pool + "/" + ec_image
+            io_total = kw["config"]["pool_based"]["source"]["rep_pool_config"][
+                "io_total"
+            ]
+            log.info("running test for Journal Pool based mirroring")
+        elif key == "image_based":
+            log.info(kw["config"])
+            for key, value in kw["config"]["image_based"].items():
+                if key == "source":
+                    kw["config"].update(value)
+                    rbd_mirror_config(**kw)
+                    for key in kw["config"]["image_based"]["source"]:
+                        kw["config"].pop(key)
+                    log.info(kw["config"])
+                    kw["config"].pop("ec-pool-k-m")
+                elif key == "destination":
+                    kw["config"].update(value)
+                    rbd_mirror_config(**kw)
+                    for key in kw["config"]["image_based"]["destination"]:
+                        kw["config"].pop(key)
+                    log.info(kw["config"])
+                    kw["config"].pop("ec-pool-k-m")
+            src_rep_pool = kw["config"]["image_based"]["source"]["rep_pool_config"][
+                "pool"
+            ]
+            rep_image = kw["config"]["image_based"]["source"]["rep_pool_config"][
+                "image"
+            ]
+            src_rep_spec = src_rep_pool + "/" + rep_image
+            dest_rep_pool = kw["config"]["image_based"]["destination"][
+                "rep_pool_config"
+            ]["pool"]
+            dest_rep_spec = dest_rep_pool + "/" + rep_image
+            src_ec_pool = kw["config"]["image_based"]["source"]["ec_pool_config"][
+                "pool"
+            ]
+            src_data_pool = kw["config"]["image_based"]["source"]["ec_pool_config"][
+                "data_pool"
+            ]
+            ec_image = kw["config"]["image_based"]["source"]["ec_pool_config"]["image"]
+            src_ec_spec = src_ec_pool + "/" + ec_image
+            dest_ec_pool = kw["config"]["image_based"]["destination"]["ec_pool_config"][
+                "pool"
+            ]
+            dest_data_pool = kw["config"]["image_based"]["source"]["ec_pool_config"][
+                "data_pool"
+            ]
+            dest_ec_spec = dest_ec_pool + "/" + ec_image
+            io_total = kw["config"]["image_based"]["source"]["rep_pool_config"][
+                "io_total"
+            ]
+            log.info("running test for Journal image based mirroring")
+        else:
+            break
+
+        try:
+            log.info("Running test on Replicated pool")
+
+            mirror2.wait_for_replay_complete(src_rep_spec)
+
+            run_fio_and_migrate_image(
+                rbd1,
+                mirror1,
+                rep_image,
+                src_rep_spec,
+                dest_rep_pool,
+                dest_rep_spec,
+                io_total,
+            )
+            if mirror1.image_exists(dest_rep_spec):
```

Comment:
can we optimize these validations for rep pool and ec pool in above `run_fio_and_migrate_image`?

Reply:
done.